dask-demo/demo.py

```diff
@@ -0,0 +1,266 @@
+from copy import deepcopy
+
+import numpy as np
+import netCDF4
+
+from activestorage import Active
+
+
+class NetCDFArray:
+    """An array stored in a netCDF file.
+
+    Supports active storage operations.
+
+    This object has been derived from `cfdm.NetCDFArray`. In order to
+    make the code easier to understand, it has been simplified from
+    its source, e.g. it doesn't support netCDF groups and doesn't
+    properly deal with string data types. This functionality would all
+    be reinstated in an operational version.
+
+    """
+
+    def __init__(self, filename=None, ncvar=None):
+        self.filename = filename
+        self.ncvar = ncvar
+
+        nc = netCDF4.Dataset(self.filename, "r")
+        v = nc.variables[self.ncvar]
```

Comment:
What happens if the keyword ncvar is not present? Would it be better to pass a NetCDF dataset (already instantiated) in along with a variable name (both required)?

Reply:
(There would be some consequences. I would work through them, but you will have seen I have a git problem right now.)

Reply:
You'll get an exception with no `ncvar` or no `filename`. I don't think that it makes sense to pass in a `netCDF.Dataset` instance, as on principle we don't want those hanging around as open file handles, but we could make the `__init__` args positional only. I'll change that.